frontend/src/features/admin-form/create/builder-and-design/MagicFormBuilderModal/MagicFormBuilderModalContent/PdfDetailsScreen.tsx

```diff
@@ -0,0 +1,162 @@
+  const [pdfFileText, setPdfFileText] = useState<string>('')
+
+  function fileToArrayBuffer(file: File) {
+    return new Promise((resolve, reject) => {
+      const reader = new FileReader()
+      reader.onload = (event) => resolve(reader.result)
+      reader.onerror = (error) => reject(error)
+      reader.readAsArrayBuffer(file)
+    })
+  }
+
+  const handleFileUpload = async (file: any) => {
+    setPdfFile(file)
+    if (file) {
+      const arrayBuffer = await fileToArrayBuffer(file)
+      try {
+        const text = await pdfToText(arrayBuffer)
+        setPdfFileText(text)
+        setValue('pdfFileText', text)
+      } catch (e) {
+        console.error(e)
+      }
+    } else {
+      setPdfFileText('')
+      setValue('pdfFileText', '')
+    }
+  }
+  async function pdfToText(data: any) {
+    const loadingTask = pdfjs.getDocument(data)
+    const pdf = await loadingTask.promise
+
+    let combinedText = ''
+
+    for (let i = 1; i <= pdf.numPages; i++) {
+      const page = await pdf.getPage(i)
+      const content = await page.getTextContent()
+      const text = content.items.map((item: any) => item.str).join(' ')
+      combinedText += text + '\n\n' // Adding two new lines to separate pages.
+    }
+    return combinedText.trim()
+  }
+
+  return (
+    <>
+      <ModalHeader color="secondary.700">
+        <Container maxW={'42.5rem'} p={0}>
+          Upload a PDF
+        </Container>
+      </ModalHeader>
+      <ModalBody whiteSpace="pre-wrap">
+        <Container maxW={'42.5rem'} p={0}>
+          <FormControl isInvalid={!!errors.pdfFileText}>
+            <FormLabel>
+              Upload a PDF - The PDF should not be a scanned copy and should not
+              contain any restricted or sensitive information.
+            </FormLabel>
+            <Controller
+              rules={{
+                validate: (pdfFileText) => {
+                  if (pdfFileText) return true
+                  return 'This PDF file cannot be processed. Please ensure that the PDF uploaded contains selectable text and is not a scanned copy.'
+                },
+              }}
+              name="pdfFileText"
+              control={control}
+              render={() => (
+                <Attachment
+                  isRequired
+                  maxSize={MAX_FILE_SIZE}
+                  accept={VALID_EXTENSIONS}
+                  showFileSize
+                  title={'Upload a PDF'}
+                  onChange={handleFileUpload}
+                  name="pdfFile"
+                  value={pdfFile}
+                  onError={(message) => setError('pdfFileText', { message })}
+                />
+              )}
+            />
+
+            <FormErrorMessage alignItems="top">
+              {!pdfFileText && <FormErrorIcon h="1.5rem" as={BxsErrorCircle} />}
+              {!pdfFileText &&
+                (pdfFile
+                  ? `${errors.pdfFileText?.message}`
+                  : 'Please upload a PDF.')}
```

Comment:
These could be combined, I think. 
```suggestion
              {!pdfFileText && <>
              <FormErrorIcon h="1.5rem" as={BxsErrorCircle} />
                (pdfFile
                  ? `${errors.pdfFileText?.message}`
                  : 'Please upload a PDF.')
                  </>
                  }
```